### ISSUE TITLE
Inner Product between vector and bivector

### ISSUE THREAD

**Issue by mw95710** [chakravala/Grassmann.jl#98]
Using @basis S"+++"

Define:

x = v1 + v2 + v3
i = v23

when I want the inner product I get x | i = 0, even though I am expecting -v2 + v3. Am I missing something?

**Comment by chakravala** [comment]
Yes, you are missing something, what you don't understand is that there are 8 possible variations of inner product, all of which are essentially the same, except for a few swaps. In `Grassmann` I adhere to the OG (original Grassmann` definitions, while in some of the literature, such as by Cambridge university authors, you may find incorrect inner products used not based on the original.

**Comment by mw95710** [comment]
I see. What I missed were the left and right contraction definitions. Just making sure, are the implementations in Grassmann.jl the same as described in the Wikipedia article?

{\displaystyle C\;\rfloor \;D:=\sum _{r,s}\langle \langle C\rangle _{r}\langle D\rangle _{s}\rangle _{s-r}}   (the left contraction)
{\displaystyle C\;\lfloor \;D:=\sum _{r,s}\langle \langle C\rangle _{r}\langle D\rangle _{s}\rangle _{r-s}}{\displaystyle C\;\lfloor \;D:=\sum _{r,s}\langle \langle C\rangle _{r}\langle D\rangle _{s}\rangle _{r-s}}   (the right contraction)

**Comment by mw95710** [comment]
I see. I was not aware of left and right contraction definitions. Just making sure, are the left and right contraction implementations in Grassmann.jl the same as in the Wikipedia article?

[https://en.wikipedia.org/wiki/Geometric_algebra](url)

**Comment by chakravala** [comment]
> are the left and right contraction implementations in Grassmann.jl the same as in the Wikipedia article?

Grassmann.jl is its own reference standardization, based on my interpreting the OG (original Grassmann) knowledge. Wikipedia does not generate a full reference implementation of these definitions, so I am not sure what their definition is. Some other sources such as authors from Cambridge use definitions that are incorrect and not compatible with the OG (original Grassmann)

**Comment by Orbots** [comment]
@mw95710 
you can use \\intprod  `⨽`, which is close ( maybe the same ) as what you expect.

As @chakravala stated, they use an alternate definition for inner product.  This is not unusual.
Kind of a GA gotcha, lots of variations out there for metric operators ( dual, innerproduct, vee, etc ).  You need to look up the definition for such operators used for each GA library you use to make sure you are on the same page.
